Improved the UI of DaoGen_Tool

supragyan opened the pull request:
Modified the below files to improve the UI of DaoGen_Tool

  **In Form1**
- Form1.cs[Design]
- Form1.cs
- Form1.Resx
- Form1.ja-JP.resx

**In Form2**
- Form2.cs[Design]
- Form2.cs
- Form2.Resx
- Form2.ja-JP.resx

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.cs:
```diff
@@ -2276,5 +2284,36 @@ private string RM_GetString(string key)
+            Button btnLoadColor = (Button)sender;
+            if (btnLoadColor.Enabled)
+            {
+                btnLoadColor.BackColor = Color.FromArgb(((int)(((byte)(91)))), ((int)(((byte)(155)))), ((int)(((byte)(213)))));
```
@supragyan 
@daisukenishino 

This program includes waste.
+ Original data type of constant, such as *91*, *155*, *213*, is *integer*.
+ The data type is converted to *byte*.
+ The converted data type is converted to *integer* again.

The following program is more simple and yields better performance:
```c#
btnLoadColor.BackColor = Color.FromArgb(91, 155, 213);
```

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/CmnMethods.cs:
```diff
@@ -45,7 +45,10 @@ public class CmnMethods
+
         /// <summary>スキーマ情報の表示</summary>
         /// <param name="table">スキーマ情報</param>
         /// <param name="writeLine">線を低か引かないか</param>
```
@daisukenishino 

「低か」→「引くか」

daisukenishino commented:
@supragyan 

### $this.Text in Form2.resx, Form2.ja-JP.resx
```
Master builder: D layer automatic generation tool- D layer definition information file generation
ja-JP : 棟梁：D層自動生成ツール（墨壺） - D層、Dao・SQLファイル生成

```
▼  It has been modified as follows.
```
D layer automatic generation tool
ja-JP : Dレイヤーの自動生成ツール
```
▼ But, I want you to modified as follows.
```
Dao automatic generation tool
ja-JP : Dao自動生成ツール
```

### Wrong translation
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-b81b6a4be01bd57cf4a1d2e2046052eeR142

- Wrong : 負荷
- Right : **読み込み** or ロード

### Possibility of DeadCode
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-c2fbd5e72c74706ca44b293efaa7d621R50

There is a possibility that the following code is **DeadCode**. Please check.
And, use the **XML comment** to class and class member (methods, properties, field).
```CS
//Sets date provider values
public static bool enableDbProviderValues;
```

supragyan commented:
@daisukenishino 

>But, I want you to modified as follows.
Wrong translation

OK.We will modify accordingly.

>And, use the XML comment to class and class member (methods, properties, field).

We will do accordingly

>There is a possibility that the following code is DeadCode. Please check.

No.This code is being used to store data provider selected value like Sql Server Client,DB2.Net,..etc
Such as **For Sql Server Client,**
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-92c78f6ec15ae4596bdd61429a5b0e15R296
**For OLEDB**
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-92c78f6ec15ae4596bdd61429a5b0e15R311

Added same for other db providers also.

### Purpose of the code
Selected data provider value from Form1 is lost for Form2 Options tab.
Because Options tab is dynamically added in runtime when click the Options button in Form2.
So we have stored Form1 selected data provider value to static field enableDbProviderValues as follows:
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-92c78f6ec15ae4596bdd61429a5b0e15R296

And assigned static field enableDbProviderValue value to data provider radio buttons in Form2 such as:
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-0f21d148be8d2fb12238fde55fa85c35R935

Please review and let us know.

daisukenishino commented:
@supragyan 

>OK. We will modify accordingly.

Let me correct to my previous post regarding the $this.Text. 
```
D layer automatic generation tool
ja-JP : D層自動生成ツール
```
This was also wrong translation.

daisukenishino commented:
@supragyan 

### Additional request
Input side should be described as a template.

https://github.com/OpenTouryoProject/OpenTouryo/issues/166#issuecomment-229870626
```XML
<data name="lblEncodingDao.Text" xml:space="preserve">
  <value>エンコーディング（Dao）</value>
</data>
```
▼ ▼ ▼
```XML
<data name="lblEncodingDao.Text" xml:space="preserve">
  <value>エンコーディング（Dao template）</value>
</data>
```

Please deal SQL side as well.
( I had not found SQL side. )

daisukenishino commented:
@supragyan

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-234199143

I added a request.

supragyan commented:
@daisukenishino 

>I added a request.
Please check.

I see.

supragyan commented:
@daisukenishino 

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-234139635
**$this.Text in Form2.resx, Form2.ja-JP.resx**
But, I want you to modified as follows.
```CS
Dao automatic generation tool
ja-JP : Dao自動生成ツール
```

Do we need to implement same as above in **$this.Text in Form1.resx, Form1.ja-JP.resx** ?
Please confirm?

daisukenishino commented:
@supragyan 

>Do we need to implement same as above in $this.Text in Form1.resx, Form1.ja-JP.resx ?

Of course.
Please implemented as follows.
#### $this.Text in Form1.resx
```
Dao automatic generation tool
```
#### $this.Text in Form1.ja-JP.resx
```
Dao自動生成ツール
```

daisukenishino commented:
@supragyan 

### Encoding (xxx template）,  エンコーディング (xxx template）
- Incorrect modification
  - lblOutputEncDao.Text  ---> エンコーディング (Dao template）  
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-cbef86441cdbf38a56a1e768dc8dea1dR254
    - Wrong : エンコーディング (Dao template）
    - Right : エンコーディング (Dao）
- There is no "エンコーディング (SQL template）".
- Form2.resx was not modified.  
English-language version of Form2.ja-JP.resx ( = Form2.resx) has not been modified.

### Purpose of the code

>Selected data provider value from Form1 is lost for Form2 Options tab.
Because Options tab is dynamically added in runtime when click the Options button in Form2.
So we have stored Form1 selected data provider value to static field enableDbProviderValues as follows:
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-92c78f6ec15ae4596bdd61429a5b0e15R296

>And assigned static field enableDbProviderValue value to data provider radio buttons in Form2 such as:
https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-0f21d148be8d2fb12238fde55fa85c35R935

If you navigate to the **Form2**, one of the check box for the data provider has become true.
If only **"this.strDAP"** exist in **Form2**, there is no problem.
Therefore, **CmnMethods.enableDbProviderValues** is not required.

supragyan commented:
@daisukenishino 

>Encoding (xxx template）, エンコーディング (xxx template）

We have modified in https://github.com/OpenTouryoProject/OpenTouryo/commit/89c310e9665f30d5bbb7aaff8a3d9b5db1345579

Please review.

supragyan commented:
@daisukenishino 

Please review this pull request and merge.
So that we will proceed to merge for this task.

daisukenishino commented:
@supragyan 

### Modify is not completed.
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-b81b6a4be01bd57cf4a1d2e2046052eeR195
```CS
<data name="btnLoadColumn.Text" xml:space="preserve">
    <value>読み込み or ロード</value>
</data>
```
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135/files#diff-cbef86441cdbf38a56a1e768dc8dea1dR480
```CS
<data name="$this.Text" xml:space="preserve">
    <value>Dao自動生成ツール</value>
</data>
```

### There is no consistency in the name of control.
```XML
<data name="lblSql.Text" xml:space="preserve">
    <value>エンコーディング
(SQL template）</value>
```
▼ ▼ ▼
```XML
<data name="lblEncodingSql.Text" xml:space="preserve">
    <value>エンコーディング
(SQL template）</value>
```

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/CmnMethods.cs:
```diff
@@ -45,7 +45,7 @@ public class CmnMethods
+
         /// <summary>スキーマ情報の表示</summary>
         /// <param name="table">スキーマ情報</param>
         /// <param name="writeLine">線を低か引かないか</param>
```
@daisukenishino 

「低か」→「引くか」

daisukenishino replied:
@yusukemaegawa 

Later, I fix this by edit button directly.

supragyan commented:
@daisukenishino 
>Modify is not completed.

Please explain more on this.

><data name="btnLoadColumn.Text" xml:space="preserve">
    <value>読み込み or ロード</value>
</data>
<data name="$this.Text" xml:space="preserve">
    <value>Dao自動生成ツール</value>
</data>
We have modified btnLoadColumn.Text to 読み込み or ロード and this.Text to Dao自動生成ツール as per https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-234139635

Please confirm



We have modified

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.Designer.cs:
```diff
@@ -60,9 +51,24 @@ private void InitializeComponent()
             this.cmbEncoding = new System.Windows.Forms.ComboBox();
+            this.btnDaoDefinitionGen = new System.Windows.Forms.Button();
+            this.pictureBox1 = new System.Windows.Forms.PictureBox();
+            this.pictureBox2 = new System.Windows.Forms.PictureBox();
```
@supragyan 
@daisukenishino 

Please set *PictureBox.BackgroundImageLayout* property to *Center*.

Note:
Currently, this property is set to *Tile*.
Then, the image is tiled across the control's client rectangle.
See the following screen capture.
![image](https://cloud.githubusercontent.com/assets/9585990/17318950/436946f2-58c5-11e6-91eb-121ba432ef38.png)

daisukenishino replied:
@yusukemaegawa 

I see.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Radiobuttons should be left-justified.
![image](https://cloud.githubusercontent.com/assets/9585990/17319435/9488931e-58c8-11e6-9a21-fe8de54b3642.png)

daisukenishino commented:
@supragyan 

>Please explain more on this.

Very troublesome.

# btnLoadColumn.Text
### en
```XML
<data name="btnLoadColumn.Text" xml:space="preserve">
    <value>Load</value>
</data>
```

### ja-JP
```XML
<data name="btnLoadColumn.Text" xml:space="preserve">
    <value>読み込み or ロード</value>
</data>
```
▼ ▼ ▼
```XML
<data name="btnLoadColumn.Text" xml:space="preserve">
    <value>ロード</value>
</data>
```

# $this.Text
### en
```XML
<data name="$this.Text" xml:space="preserve">
    <value>D layer automatic generation tool</value>
</data>
```
### ja-JP
```XML
<data name="$this.Text" xml:space="preserve">
    <value>Dao自動生成ツール</value>
</data>
```
▼ ▼ ▼
```XML
<data name="$this.Text" xml:space="preserve">
    <value>D層自動生成ツール</value>
</data>
```

yusukemaegawa commented:
@supragyan 
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236815944

And, please equally space the radio buttons.

supragyan commented:
@daisukenishino 
@yusukemaegawa 

Ok we will check and modify.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
+    <value>25, 13</value>
+  </data>
+  <data name="lblFamilyName.Text" xml:space="preserve">
+    <value>姓：</value>
```
@supragyan 
@daisukenishino 

In [Form2.resx](https://github.com/SymphonyTeleca/OpenTouryo/blob/9bf46a7d9cbf4fbfa88f4ca09ad45faaec8f2324/root/programs/C%23/Frameworks/Tools/DaoGen_Tool/Form2.resx#L1664), this value is marked with an asterisk.
```xml
<data name="lblFamilyName.Text" xml:space="preserve">
  <value>*Last Name</value>
</data>
```

But, in this file, this value is *NOT* marked with an asterisk.
What is the difference?

daisukenishino replied:
@yusukemaegawa 

I'll confirm this specification.

daisukenishino replied:
@yusukemaegawa 
@supragyan 

This asterisk is the meaning of the optional.
But I think that usually means required entry.
Result of discussion, it was decided to change as follows.

### GivenName and FamilyName
Change font color of this part to dark gray after remove *.
Note: Both of Form2.resx and Form2.ja-JP.resx are the target.

### Options
Remove * from "Selected".
This label will be displayed when the option has been entered in the options screen.

supragyan replied:
@daisukenishino 

>This label will be displayed when the option has been entered in the options screen.
This label is always displayed.

Do you want us to change as below functionality:

- On Load label visibility is set to false.
- On clicking Options label visibility is set to true.

Please confirm.

daisukenishino replied:
@supragyan 

>- On Load, label visibility is set to false.

Ok.

>- On clicking Options button, label visibility is set to true.

No.
**On clicking OK button in Options screen**, label visibility is set to true.

daisukenishino replied:
@supragyan 

There are two additional requests.

### 1 request
Add tooltip to GivenName and FamilyName.

Wording of the tool tip is
- en(default) : This field is optional.
- ja-JP : この項目は省略可能です。

### 2 request
Please change the "Selected" to the "Options were specified.".
- en (default) : Options were specified.
- ja-JP : オプションが指定されました。

supragyan replied:
@daisukenishino 

>Add tooltip to GivenName and FamilyName.
Ok. We will implement tooltip on GivenName and FamilyName. label controls.

>Please change the "Selected" to the "Options were specified.".
Ok.We will implement accordingly.

supragyan replied:
@daisukenishino 

>Add tooltip to GivenName and FamilyName.
Please change the "Selected" to the "Options were specified.".

We have implemented as above and did code commit in [#135](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/ecdfb3191f321fea9d4689494855f970dfaaa117) pull request.


Please review.

supragyan replied:
@daisukenishino 
Please review this pull reuest and let us know.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.resx:
```diff
@@ -835,153 +715,659 @@
-  </data>
-  <data name="&gt;&gt;Restart_ToolStripMenuItem.Type" xml:space="preserve">
-    <value>System.Windows.Forms.ToolStripMenuItem, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+    <value>D layer automatic generation tool</value>
```
@supragyan 
@daisukenishino 

Please revise as follows:
- D layer -> Data tier
- automatic generation tool -> generator

daisukenishino replied:
@yusukemaegawa 

I desire to maintain the status quo.
I think the above modification also extends to other parts of the body, such as the document.
Therefore, I think that we should register this problem to another issue (e.g. https://github.com/OpenTouryoProject/OpenTouryo/issues/149).

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Labels should be right-justified.
TextBoxes and ComboBoxes should be left-justified.
![image](https://cloud.githubusercontent.com/assets/9585990/17320985/397e6c38-58d1-11e6-8871-655aac0b5ee3.png)

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
-  <data name="groupBox3.Text" xml:space="preserve">
-    <value>テーブルメンテナンス画面を自動生成する（エンコーディングはDaoと同じになる</value>
+  <data name="cbxTSIndisp.Size" type="System.Drawing.Size, System.Drawing">
+    <value>252, 30</value>
```
@supragyan 
@daisukenishino 

Please set *cbxTSIndisp.TextAlign* property to *MiddleLeft*.

daisukenishino replied:
@yusukemaegawa

Next week, I would like to heare your explanation about this.

daisukenishino replied:
@supragyan 

Please set cbxTSIndisp.TextAlign property to MiddleLeft.
(see : DaoGen_Tool_UI_(xxxxxxxx)20160609.pdf)

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.Designer.cs:
```diff
@@ -29,52 +29,51 @@ protected override void Dispose(bool disposing)
+            this.cmbDDEncoding = new System.Windows.Forms.ComboBox();
+            this.lblInputEncoding = new System.Windows.Forms.Label();
             this.rbnDTL_VB = new System.Windows.Forms.RadioButton();
+            this.lblLanguage = new System.Windows.Forms.Label();
```
@supragyan 
@daisukenishino 

I think that *Language* and *Last Name / Given Name* are categorized to *Output settings*.

supragyan replied:
@daisukenishino 
@yusukemaegawa 

>I think that Language and Last Name / Given Name are categorized to Output settings.

But in DaoGen_Tool_UI ppt, Language and Last Name / Given Name are categorized to Input Settings.

![capture](https://cloud.githubusercontent.com/assets/9566037/17355198/6f77efd0-596c-11e6-83d3-f570e65a24b7.PNG)

So we have implemented in Input Settings.
Please confirm on this.

daisukenishino replied:
@yusukemaegawa 

- Input settings and input information for the input is in the input group.
- Only output settings is in the output group.

Therefore, there is no problem about the location of "last name" and "given name".

yusukemaegawa commented:
@supragyan 
@daisukenishino 

It may be difficult for beginners to understand how to use this tool.
So, it might be a good idea to add link to help site.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
-    <value>Daoファイル出力のエンコーディングを選択する</value>
+  <data name="lblSql.Text" xml:space="preserve">
+    <value>エンコーディング
+(SQL template）</value>
```
@supragyan 
@daisukenishino 

"(" is a *half-size* character.
"）" is a *full-size* character.

So, please revise as follows:
```
(SQL template)
```

daisukenishino replied:
@supragyan 

My description is incorrect.
Use the half-size characters to both.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
+  </data>
+  <data name="lblEncodingDao.Text" xml:space="preserve">
+    <value>エンコーディング
+(Dao template）</value>
```
@supragyan 
@daisukenishino 

"(" is a *half-size* character.
"）" is a *full-size* character.

So, please revise as follows:
```
(Dao template)
```

daisukenishino replied:
@supragyan 

Same as above.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.ja-JP.resx:
```diff
@@ -112,61 +112,175 @@
-  <data name="btnLoadColumn.Text" xml:space="preserve">
-    <value>列情報のロード </value>
+  <data name="lblSchemaInfo.Text" xml:space="preserve">
+    <value>スキーマ情報</value>
```
@daisukenishino 
@supragyan 

It might be a good idea to remove *Schema Information*.

I don't understand what *Schema Information* helps users to generate data tier programs.

daisukenishino replied:
@yusukemaegawa 

So it is necessary for debugging, I want to leave this function.
But, certainly, I think that it is good to move this function to debug window.
But, this task should be registered as a new issue.
Because start of merge task is delayed by the end of this task is delayed.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.cs:
```diff
@@ -156,8 +156,8 @@ private void Form1_Load(object sender, EventArgs e)
             this.btnDelTable.Enabled = false;
```
@supragyan 
@daisukenishino 

Even if *btnDelTable.Enabled* property changes, *btnDelTable.BackColor* property does not change.
- btnDelTable
    - Enabled
        - BackColor: #5B9BD5
        - ForeColor: White
    - Disabled
        - BackColor: *#5B9BD5* (*unchanged*)
        - ForeColor: DarkGray
- btnLoadColumn
    - Enabled
        - BackColor: #5B9BD5
        - ForeColor: White
    - Disabled
        - BackColor: *LightGray* (*changed*)
        - ForeColor: DarkGray

daisukenishino replied:
@yusukemaegawa 

Next week, I would like to heare your explanation about this.

daisukenishino replied:
@supragyan 

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73269022

In summary.


If btnDelTable.Enabled == false Then 
- BackColor: LightGray (changed)
- ForeColor: DarkGray

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.resx:
```diff
@@ -835,153 +715,659 @@
+    <value>True</value>
+  </data>
+  <data name="label5.Font" type="System.Drawing.Font, System.Drawing">
+    <value>Microsoft Sans Serif, 8.3pt</value>
```
@supragyan 
@daisukenishino 

The font family of *label5* is inconsistent with other labels.
![image](https://cloud.githubusercontent.com/assets/9585990/17351486/ef2792ba-5969-11e6-92ad-12be8c377c38.png)

daisukenishino replied:
@supragyan 

Please unified the font.

supragyan replied:
@daisukenishino 
@yusukemaegawa 

>The font family of label5 is inconsistent with other labels.

Now the font of **Selected Tables**,**Column Info** and **Primary key** labels is same i.e **Microsoft Sans Serif**
Do we need to change as below:
- Selected Tables
 -  **Microsoft Sans Serif**(unchanged)
- Column Info
 - **MS UI Gothic**
- Primary key
 - **MS UI Gothic**

Please confirm.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.resx:
```diff
@@ -835,153 +715,659 @@
+    <value>NoControl</value>
+  </data>
+  <data name="label5.Location" type="System.Drawing.Point, System.Drawing">
+    <value>10, 16</value>
```
@supragyan 
@daisukenishino 

The labels should be left-justified.
![image](https://cloud.githubusercontent.com/assets/9585990/17351505/3763bae0-596a-11e6-8196-4639506007da.png)

daisukenishino replied:
@supragyan 

Please be left-justified.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
-  <data name="gbxDaoTemplateLanguage.Text" xml:space="preserve">
-    <value>言語を選択する</value>
+  <data name="lblSetSourceTemplate.Text" xml:space="preserve">
+    <value>ソース テンプレート ファイル</value>
```
@supragyan 
@daisukenishino 

User should specify the *folder* which stores *source template files*.
So, please revise as follows:
- English
```xml
<value>Source Template Folder</value>
```

- Japanese
```xml
<value>ソース テンプレート フォルダ</value>
```

daisukenishino replied:
@yusukemaegawa 

I think it is appropriate.

@supragyan 

https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73271259
Fix as described above.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Please revise [Form2.resx](https://github.com/SymphonyTeleca/OpenTouryo/blob/9bf46a7d9cbf4fbfa88f4ca09ad45faaec8f2324/root/programs/C%23/Frameworks/Tools/DaoGen_Tool/Form2.resx#L250).
```xml
<value>Skip the first line (When there is header)</value>
```

yusukemaegawa commented:
@supragyan 
@daisukenishino 

The line is cut in half lengthwise.
![image](https://cloud.githubusercontent.com/assets/9585990/17353876/0f1defbe-597f-11e6-8741-cb8d8dc3464e.png)

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Users have already selected data provider in *STEP 1*.
Therefore, *Select Data Provider* section is unnecessary in option window.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.Designer.cs:
```diff
@@ -501,178 +514,228 @@ private void InitializeComponent()
+            this.BackColor = System.Drawing.Color.White;
+            this.Controls.Add(this.panel1);
+            this.ForeColor = System.Drawing.SystemColors.ControlText;
             this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
```
@supragyan 
@daisukenishino 

If user can not resize this window, it might be a good idea to set *Form2.MaximizeBox* property to *false*.

daisukenishino replied:
@yusukemaegawa

Next week, I would like to heare your explanation about this.

daisukenishino replied:
@supragyan 

If this window is not resizable.
```CS
this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
```
Set *Form2.MaximizeBox* property to *false*.
```CS
this.MaximizeBox = false;
```

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,371 @@
-  <data name="btnSetOutput.Text" xml:space="preserve">
-    <value>ルート フォルダ パスを指定</value>
+  <data name="lblSetDaoDefinition.Text" xml:space="preserve">
+    <value>D層定義情報ファイル</value>
```
@supragyan 
@daisukenishino 

Users may not understand that "D層定義情報ファイル" and "中間情報" (in STEP 1) mean the same thing.
So, please unify terms.

supragyan replied:
@daisukenishino 

>Users may not understand that "D層定義情報ファイル" and "中間情報" (in STEP 1) mean the same thing.
So, please unify terms.

Yes we understood.

If we need to unify terms then we thought "D層定義情報ファイル" is more understandable than "中間情報の生成".

Please confirm your opinion.

daisukenishino replied:
@supragyan 

### Step1
We also thought below would be more appropriate.
- **en**
Generate D layer definition file

- **ja-JP**
D層定義情報を生成

### Step2
- **en**
D layer definition file

- **ja-JP**
D層定義情報ファイル

supragyan replied:
@daisukenishino 

OK.We will modify.

yusukemaegawa commented:
@daisukenishino 

It might be a good idea to change procedure as follows:

Current procedure:
+ STEP 1
    + Select Data Provider
    + Get Table List
    + Delete unnecessary tables from table list
    + Load column information
    + Set primary key (if necessary)
    + **Generate temp info**
    + Go to STEP 2
+ STEP 2
    + **Load temp info**
    + Configure input / output settings
    + Generate program

Changed procedure:
+ STEP 1
    + Select Data Provider
    + Get Table List
    + Delete unnecessary tables from table list
    + Load column information
    + Set primary key (if necessary)
    + Go to STEP 2  
    (**It is necessary to store table information.**)
+ STEP 2
    + Configure input / output settings
    + Generate program  
    (**Temp info is generated internally.**)

Users do not have to consider *temp info*.
So, users do not have to generate *temp info*, and do not have to set encoding of *temp info*.

Please review.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.cs:
```diff
@@ -507,7 +507,7 @@ private void button1_Click(object sender, EventArgs e)
             //this.tabPage3.Text = "タブ３";
             this.tabPage3.Text = this.RM_GetString("TabPage3");
-            
+
```
@daisukenishino 
@supragyan 

If these tabs are necessary in case of debug only, it might be  a good idea to switch *Form1.FormBorderStyle* property.
```c#
if (System.Diagnostics.Debugger.IsAttached)
{
    // デバッグ実行のときは、リサイズ可とする
    this.FormBorderStyle = FormBorderStyle.Sizable;
}
else
{
    // デバッグなしで実行するときは、リサイズ不可とする
    this.FormBorderStyle = FormBorderStyle.FixedSingle;
}
```

daisukenishino replied:
@yusukemaegawa

Next week, I would like to heare your explanation about this.

daisukenishino replied:
@supragyan 

Add the following code to [Form1.Form1_Load](https://github.com/SymphonyTeleca/OpenTouryo/blob/9bf46a7d9cbf4fbfa88f4ca09ad45faaec8f2324/root/programs/C%23/Frameworks/Tools/DaoGen_Tool/Form1.cs#L140)
```CS
if (System.Diagnostics.Debugger.IsAttached || Environment.CommandLine.ToUpper() == "DBG")
{
    // デバッグ実行のときは、リサイズ可とする
    this.FormBorderStyle = FormBorderStyle.Sizable;
}
else
{
    // デバッグなしで実行するときは、リサイズ不可とする
    this.FormBorderStyle = FormBorderStyle.FixedSingle;
}
```

daisukenishino replied:
@supragyan 

Please change part of the [above](https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73826573) as follows.
```
Environment.CommandLine
```
This returns the entire command, including the executable file name.

▼ ▼ ▼
```
Environment.GetCommandLineArgs()[1]
```
This returns the command that has been splited by a space.

supragyan replied:
@daisukenishino 

>Please change part of the above as follows.

We have modified as per above in [#135](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/6a72e70f39f48f0282ce2093a8ea918b8c75386e#diff-92c78f6ec15ae4596bdd61429a5b0e15R159)

Please check.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.Designer.cs:
```diff
@@ -38,20 +38,11 @@ private void InitializeComponent()
             this.menuStrip1 = new System.Windows.Forms.MenuStrip();
-            this.Restart_ToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
-            this.Close_ToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
+            this.STEP1ToolStripMenuItem = new System.Windows.Forms.ToolStripMenuItem();
```
@supragyan 
@daisukenishino 

Currently, this control has *no menu items*.
Therefore, this control should be *Label control*.

daisukenishino replied:
@yusukemaegawa 

Next week, I would like to heare your explanation about this.

daisukenishino replied:
@supragyan 

Change the ToolStripMenuItem control to Label control.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Each window, such as *STEP 1* or *STEP 2* or *OPTION*, has a different *BackColor*.
It might be a good idea to unify this property.
- *STEP 1* : Control
- *STEP 2* : Transparent
- *OPTION* : #BDD7EE

daisukenishino commented:
@yusukemaegawa 
Thank you a lot of points out.
@supragyan

daisukenishino commented:
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73100169
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236815944
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236816357
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236828993
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73664636
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73664732
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73665505
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73665610
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237135506

supragyan commented:
@daisukenishino 

Thank you for confirmation.

We will modify accordingly.

daisukenishino commented:
@yusukemaegawa 

https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237155595

There is a case to use only the features of Form2 using a pre-generated D layer definition information file.
Therefore, this became a regrettable proposal.

daisukenishino commented:
@yusukemaegawa 
@supragyan 

### Checking
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73663046
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73669673
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73665400
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73668488
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73666528
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73666438

### Interrupted
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73661065
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73662013
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73267487
- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237809538

daisukenishino commented:
@yusukemaegawa 

This fix contains #176.
Please check https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/a737ffa8e57177ad2cdd83c154003936401e865e.

daisukenishino commented:
@supragyan 

>### Checking

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73663046

・・・

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73669673

https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73821446

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236837524

Add Link to [5_User_Guide(D_Layer_Automatic_Generation)_EN.doc](https://github.com/OpenTouryoProject/OpenTouryoDocuments/blob/master/documents/1_User_Guide/en/5_User_Guide(D_Layer_Automatic_Generation)_EN.doc) by LinkLabel control at Right side of Form1 header of Step1 (if ja-JP then [5_User_Guide(D_Layer_Automatic_Generation).doc](https://github.com/OpenTouryoProject/OpenTouryoDocuments/blob/master/documents/1_User_Guide/ja-JP/5_User_Guide(D_Layer_Automatic_Generation).doc)).


>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73665400
>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237117942
>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73668488
>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73666528
>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73666438

daisukenishino commented:
@supragyan
@yusukemaegawa 

>### Interrupted

We listed the results of the verification at the following.

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73661065

https://github.com/OpenTouryoProject/OpenTouryo/issues/149#issuecomment-238152818

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73662013

No change.

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73267487

https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73826573

>- https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237809538

https://github.com/OpenTouryoProject/OpenTouryo/issues/149#issuecomment-238152818

supragyan commented:
@daisukenishino 

We have modified each comment in this pull request.

Please review.

supragyan commented:
@daisukenishino 
Please review this pull request.
So that we will restart merge of Data Maintainance and this DaoGen tool changes simultaneously from root.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form2.ja-JP.resx:
```diff
@@ -112,205 +112,422 @@
-  <data name="label2.Size" type="System.Drawing.Size, System.Drawing">
-    <value>61, 13</value>
+  <data name="cbxTSIndisp.TextAlign" type="System.Drawing.ContentAlignment, System.Drawing">
+    <value>MiddleLeft</value>
```
@supragyan 
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73821446

Please revise [Form2.resx](https://github.com/SymphonyTeleca/OpenTouryo/blob/21078eb5f39c4c6592af634d70d1ddbc12498869/root/programs/C%23/Frameworks/Tools/DaoGen_Tool/Form2.resx#L1015) too.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-238147901

> ```XML
> <data name="cbxDaoDefinitionHeader.Text" xml:space="preserve">
>     <value>When there is Header (Skip the first line)</value>
> </data>
> ```

Please change *H* in value tag to lowercase and add an article.
```XML
<data name="cbxDaoDefinitionHeader.Text" xml:space="preserve">
    <value>When there is a header (Skip the first line)</value>
</data>
```

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.cs:
```diff
@@ -152,12 +156,23 @@ private void Form1_Load(object sender, EventArgs e)
             // コンボを初期化する。
             this.Init_cmb();
 
+            if (System.Diagnostics.Debugger.IsAttached)
```
@supragyan 
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r73826573

>@supragyan 
>
>Add the following code to [Form1.Form1_Load](https://github.com/SymphonyTeleca/OpenTouryo/blob/9bf46a7d9cbf4fbfa88f4ca09ad45faaec8f2324/root/programs/C%23/Frameworks/Tools/DaoGen_Tool/Form1.cs#L140)
>```CS
>if (System.Diagnostics.Debugger.IsAttached || Environment.CommandLine.ToUpper() == "DBG")
>{
>    // デバッグ実行のときは、リサイズ可とする
>    this.FormBorderStyle = FormBorderStyle.Sizable;
>}
>else
>{
>    // デバッグなしで実行するときは、リサイズ不可とする
>    this.FormBorderStyle = FormBorderStyle.FixedSingle;
>}
>```

Conditional expression about *command line arguments* is not implemented.
Please revise as above.

supragyan commented:
@daisukenishino 
We have modified each comment in this pull request.

Please review.

supragyan commented:
@daisukenishino

>Add tooltip to GivenName and FamilyName.
Please change the "Selected" to the "Options were specified.".

We have implemented as above and did code commit in this pull request.

Please review.

yusukemaegawa commented:
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236828993

Response to above comment is missing from [this list](https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-237814058).
We would like to get your feedback.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

I see a part of debug area.
Please move debug area more to the right.
![image](https://cloud.githubusercontent.com/assets/9585990/18042038/26181a30-6df9-11e6-9244-dffbcb6afa32.png)

yusukemaegawa commented:
@supragyan 
@daisukenishino 

Layout of option window collapses when running in my computer.
Please confirm.
![image](https://cloud.githubusercontent.com/assets/9585990/18042252/cbea7a9c-6dfa-11e6-82f6-84a3176f5404.png)

supragyan commented:
@yusukemaegawa 

The above Options screen appears as below in our computer.
![n](https://cloud.githubusercontent.com/assets/9566037/18042453/1813f9a0-6ddf-11e6-8caf-31f3ec3fb82a.png)

Please confirm.

yusukemaegawa commented:
@supragyan 

Thank you for the quick response.
What is *display dpi size* of your computer?

(Reference)How to Change the Display DPI Size in Windows 7 and Windows 8
http://www.sevenforums.com/tutorials/443-dpi-display-size-settings-change.html

*Display dpi size* of my computer is *Medium - 125% (default)*.

supragyan commented:
@yusukemaegawa 

>What is display dpi size of your computer?

Display dpi size of my computer is smaller -100% (default).

Do we need to change our dpi size to Medium - 125% (default) and modify accordingly?

Please confirm.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Form1.cs:
```diff
@@ -152,12 +156,23 @@ private void Form1_Load(object sender, EventArgs e)
             // コンボを初期化する。
             this.Init_cmb();
 
+            if (System.Diagnostics.Debugger.IsAttached || Environment.GetCommandLineArgs()[1].ToUpper() == "DBG")
```
@supragyan 
@daisukenishino 

[IndexOutOfRangeException](https://msdn.microsoft.com/en-us/library/system.indexoutofrangeexception(v=vs.110).aspx) occurs when running this tool *without command line arguments*.
So, please add the logic to check length of command line arguments.

```c#
if (System.Diagnostics.Debugger.IsAttached || (Environment.GetCommandLineArgs().Length > 1 && Environment.GetCommandLineArgs()[1].ToUpper() == "DBG"))
```

daisukenishino replied:
@supragyan 

Pleaes correspond as above.

supragyan replied:
@daisukenishino 
@yusukemaegawa 

We have modified as above in [#125](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/de96cc1e138a19524ae83a4ab2dee0c8a5d5c404#diff-92c78f6ec15ae4596bdd61429a5b0e15R159)

Please review.

daisukenishino commented:
@supragyan 

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-243030146

Please correspond to the following.
https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236828993

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-243045301

In this issue, please only correspond to the "Smaller 100% (default)".
Register the another issue to correspond Medium 125% and Larger 150%.

supragyan commented:
@daisukenishino 
>In this issue, please only correspond to the "Smaller 100% dpi size (default)".
Register the another issue to correspond Medium 125% and Larger 150% dpi size.

Ok.

daisukenishino commented:
@supragyan 

- The following correspondence is not complete.
  - https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-236828993
- The following correspondence is incorrect.  
You should give tooltip to the textbox, not label.
  - https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r76369764

supragyan commented:
@daisukenishino 

>The following correspondence is not complete.
Labels should be right-justified.
Debug area protruded.

But we already implemented as above.
And the screen appears as:
![new](https://cloud.githubusercontent.com/assets/9566037/18085347/6c3d10a4-6ec8-11e6-82ea-27c3f4364b1c.png)

>The following correspondence is incorrect.
You should give tooltip to the textbox, not label.

We have already informed [here](https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r76181946)

Ok We will implement accordingly.

supragyan commented:
@daisukenishino 

>https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-243368699

We have added ToolTip to Given Name and Family Name.

But we modified ToolTip changes as follows:
- Removed ToolTip control from Form2 because if we implement ToolTip to specific control its added in some unwanted controls in [Form2.Designer.cs](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/65d063f936a8223aac1866a57098ebee967c6ca7#diff-c55158d3413aaea15ec39ad4baca1e92L104)
- Added resource key for ToolTip text in [Resource.resx](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/65d063f936a8223aac1866a57098ebee967c6ca7#diff-490e3cde4fd952bd045c202695f4ce52R271) and [Resource.ja-JP.resx](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/65d063f936a8223aac1866a57098ebee967c6ca7#diff-e8fa0dd2efc417c26e21b3613b80f684R265)

- Added this resource key in Given Name and Family Name as [#135](https://github.com/SymphonyTeleca/OpenTouryo/pull/135/commits/65d063f936a8223aac1866a57098ebee967c6ca7#diff-0f21d148be8d2fb12238fde55fa85c35R986).

Please review and confirm.

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Resources/Resource.ja-JP.resx:
```diff
@@ -258,4 +258,10 @@
+    <value>操作はキャンセルされました。</value>
+  </data>
+  <data name="ToolTipText" xml:space="preserve">
+    <value>オプションが指定されました。</value>
```
@supragyan 
@daisukenishino 

This text should be *この項目は省略可能です。*.

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r75995581
> 
> ### 1 request
> Add tooltip to GivenName and FamilyName.
> 
> Wording of the tool tip is
> - en(default) : This field is optional.
> - ja-JP : この項目は省略可能です。

yusukemaegawa commented on root/programs/C#/Frameworks/Tools/DaoGen_Tool/Resources/Resource.resx:
```diff
@@ -264,4 +264,10 @@
+    <value>The operation was canceled.</value>
+  </data>
+  <data name="ToolTipText" xml:space="preserve">
+    <value>Options were specified.</value>
```
@supragyan 
@daisukenishino 

This text should be *This field is optional.*.

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#discussion_r75995581
> 
> ### 1 request
> Add tooltip to GivenName and FamilyName.
> 
> Wording of the tool tip is
> - en(default) : This field is optional.
> - ja-JP : この項目は省略可能です。

supragyan replied:
@yusukemaegawa 

Sorry for inconvenience.
We have modified as above.

Please confirm.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

>>The following correspondence is not complete.
>>Labels should be right-justified.
>>Debug area protruded.

>But we already implemented as above.

I see *STEP2 window* as follows:  
![image](https://cloud.githubusercontent.com/assets/9585990/18118880/8eb207fa-6f92-11e6-9ee7-61326c60fb0b.png)

Please confirm.

supragyan commented:
@daisukenishino 
@yusukemaegawa 

>I see STEP2 window as follows:

We are not able to replicate this issue.

We have tested by changing dpi size to 125% but still we are not getting above screen.
- **dpi size to 100%** 
![c2](https://cloud.githubusercontent.com/assets/9566037/18119895/838d0194-6f7a-11e6-9f3a-aa6a674d10ea.png)

- **dpi size to 125%** 
![c1](https://cloud.githubusercontent.com/assets/9566037/18119916/9777eed0-6f7a-11e6-9190-da0e14ac7b93.png)

Please confirm.

yusukemaegawa commented:
@supragyan 
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/135#issuecomment-243681657

Thank you for the confirmation.

This issue might be caused by the *difference of font*, that is, the *difference of width of each character*.

Please set *Label.AutoSize* property to *false* and *Label.TextAlign* property to *TopRight* for labels which should be right-justified.

supragyan commented:
@yusukemaegawa 
@daisukenishino 

We have modified as follows in this pull request:
- Set Label.AutoSize property to false
- Set Label.TextAlign property to TopRight 

Please check and let us know.

yusukemaegawa commented:
@supragyan 

Thank you for your response.
I could confirm that labels were right-justified.
